tensornetwork/matrixproductstates/mps.py

```diff
@@ -341,6 +431,315 @@ def measure_two_body_correlator(self, op1: Union[BaseNode, Tensor],
+    Returns:
+      scalar `Tensor`: the truncated weight of the truncation.
+    
+    """
+    if len(gate.shape) != 4:
+      raise ValueError('rank of gate is {} but has to be 4'.format(
+          len(gate.shape)))
+    if site1 < 0 or site1 >= len(self) - 1:
+      raise ValueError(
+          'site1 = {} is not between 0 <= site < N - 1 = {}'.format(
+              site1, len(self)))
+    if site2 < 1 or site2 >= len(self):
+      raise ValueError('site2 = {} is not between 1 <= site < N = {}'.format(
+          site2, len(self)))
+    if site2 <= site1:
+      raise ValueError('site2 = {} has to be larger than site2 = {}'.format(
+          site2, site1))
+    if site2 != site1 + 1:
+      raise ValueError(
+          'site2 ={} != site1={}. Only nearest neighbor gates are currently '
+          'supported'.format(site2, site1))
+
+    if (max_singular_values or
+        max_truncation_err) and self.center_position not in (site1, site2):
+      raise ValueError(
+          'center_position = {}, but gate is applied at sites {}, {}. '
+          'Truncation should only be done if the gate '
+          'is applied at the center position of the MPS'.format(
+              self.center_position, site1, site2))
+
+    gate_node = Node(gate, backend=self.backend.name)
+
+    self.nodes[site1][2] ^ self.nodes[site2][0]
+    gate_node[2] ^ self.nodes[site1][1]
+    gate_node[3] ^ self.nodes[site2][1]
+    left_edges = [self.nodes[site1][0], gate_node[0]]
+    right_edges = [gate_node[1], self.nodes[site2][2]]
+    result = self.nodes[site1] @ self.nodes[site2] @ gate_node
+    U, S, V, tw = split_node_full_svd(
+        result,
+        left_edges=left_edges,
+        right_edges=right_edges,
+        max_singular_values=max_singular_values,
+        max_truncation_err=max_truncation_err,
+        left_name=self.nodes[site1].name,
+        right_name=self.nodes[site2].name)
+    V.reorder_edges([S[1]] + right_edges)
+    left_edges = left_edges + [S[1]]
+    self.nodes[site1] = contract_between(
+        U, S, name=U.name).reorder_edges(left_edges)
+    self.nodes[site2] = V
+    self.nodes[site1][2] | self.nodes[site2][0]
+    return tw
+
+  def apply_one_site_gate(self, gate: Union[BaseNode, Tensor],
+                          site: int) -> None:
+    """
+    Apply a one-site gate to an MPS. This routine will in general 
+    destroy any canonical form of the state. If a canonical form is needed, 
+    the user can restore it using `FiniteMPS.position`
+
+    Args:
+      gate: a one-body gate
+      site: the site where the gate should be applied
+      
+    """
+    if len(gate.shape) != 2:
+      raise ValueError('rank of gate is {} but has to be 2'.format(
+          len(gate.shape)))
+    if site < 0 or site >= len(self):
+      raise ValueError('site = {} is not between 0 <= site < N={}'.format(
+          site, len(self)))
+    gate_node = Node(gate, backend=self.backend.name)
```

Comment:
Backends now support passing the actual backend object instead of just the name, so prefer to use `backend=self.backend` here.

Reply:
that's great! I wanted to actually submit an issue for that